java/dataset/src/main/java/org/apache/arrow/dataset/jni/DirectReservationListener.java

```diff
@@ -87,11 +87,40 @@ public void unreserve(long size) {
   public long getCurrentDirectMemReservation() {
     try {
       final Class<?> classBits = Class.forName("java.nio.Bits");
-      final Field f = classBits.getDeclaredField("reservedMemory");
+      final Field f;
+      Field fBaseOnJDKVersion;
```

Comment:
Nit but I would just use `f` and make it non-final here no need to complicate it with extra assignments

Reply:
Changed